retrieval_service/datastore/providers/neo4j_graph.py

```diff
@@ -83,6 +85,24 @@ async def create_amenities(tx, amenities):
                     hour=amenity.hour,
                 )
 
+                # Create Category node
+                await tx.run(
+                    """
+                    MERGE (c:Category {name: $category})
```

Comment:
By using merge it ensures if `x category node` already exists, do not create it again.

Reply:
please add this as a code comment